src/main.rs

```diff
@@ -294,85 +297,95 @@ fn main() {
         };
     }
 
-    // Instantiate cache. This will not yet create the cache directory!
-    let cache = Cache::new(platform, &config.directories.cache_dir.path);
-
-    // Clear cache, pass through
-    if args.clear_cache {
-        clear_cache(&cache, args.quiet, enable_styles);
-    }
+    let mut err_cmd: Option<String> = None;
 
-    // Cache update, pass through
-    let cache_updated = if should_update_cache(&cache, &args, &config) {
-        update_cache(&cache, args.quiet, enable_styles);
-        true
-    } else {
-        false
+    // Collect languages
+    let languages = match args.language {
+        Some(ref lang) => vec![lang.to_string()],
+        None => get_languages_from_env(),
     };
 
-    // Check cache presence and freshness
-    if !cache_updated
-        && (args.list || !args.command.is_empty())
-        && check_cache(&cache, &args, enable_styles) == CheckCacheResult::CacheMissing
-    {
-        process::exit(1);
-    }
+    for platform in platforms {
+        // Instantiate cache. This will not yet create the cache directory!
+        let cache = Cache::new(platform, &config.directories.cache_dir.path);
```

Comment:
So far, we only ever had one cache for the whole program, which makes sense, considering that we have all of the tldr-pages checked out in one directory. Right now, this code has a bug where we check whether the same directory needs to be updated (or cleared) for every platform the a user passes. I think it also lists some commands more than once, if there are multiple versions for different platforms.

I think we should stick to only having one `cache`. About adding support for multiple platforms, we could either make `Cache` have its own vector of platforms, or, just pass the platforms when they are needed. I like the latter the most, it will also lead to a solution where we don't need an owned `Vec` and don't have lifetimes on `Cache` (only in the methods that will then take `platforms: &[Platform]`).